data/data/powervs/cluster/dns/dns.tf

```diff
@@ -20,3 +20,87 @@ resource "ibm_cis_dns_record" "kubernetes_api_internal" {
   content   = var.load_balancer_int_hostname
   ttl       = 60
 }
+
+locals {
+  proxy_count = var.publish_strategy == "Internal" ? 1 : 0
+}
+
+resource "ibm_is_ssh_key" "dns_ssh_key" {
+  count      = local.proxy_count
+  name       = "${var.cluster_id}-dns-ssh-key"
+  public_key = var.ssh_key
+}
+
+resource "ibm_is_security_group" "dns_vm_sg" {
+  count = local.proxy_count
+  name  = "${var.cluster_id}-dns-sg"
+  vpc   = var.vpc_id
+}
+
+# allow all outgoing network traffic
+resource "ibm_is_security_group_rule" "dns_vm_sg_outgoing_all" {
+  count     = local.proxy_count
+  group     = ibm_is_security_group.dns_vm_sg[0].id
+  direction = "outbound"
+  remote    = "0.0.0.0/0"
+}
+
+# allow all incoming network traffic on port 22
```

Comment:
Will we need SSH for anything other than debug? Might not be worth exposing the port

Reply:
No we don't need SSH for anything other than debug.  Which I am sure will be needed in the future.  Also, all of the master and worker nodes in an OpenShift cluster have ssh enabled.

Reply:
okay we'll leave it open for now